kyo-combinators/shared/src/main/scala/kyo/Constructors.scala

```diff
@@ -65,11 +65,17 @@ extension (kyoObject: Kyo.type)
     def fromAutoCloseable[A <: AutoCloseable, S](closeable: => A < S): A < (S & Resource & IO) =
         acquireRelease(closeable)(c => IO(c.close()))
 
-    def fromEither[E, A](either: => Either[E, A]): A < Abort[E] =
-        Abort.get(either)
+    def fromEither[E, A, S](either: => Either[E, A] < S): A < (S & Abort[E]) =
+        either.map(Abort.get(_))
 
-    def fromResult[E, A](result: => Result[E, A]): A < Abort[E] =
-        Abort.get(result)
+    def fromOption[A, S](option: => Option[A] < S): A < (S & Abort[Maybe.Empty]) =
+        option.map(o => Abort.get(o.toRight[Maybe.Empty](Maybe.Empty)))
+
+    def fromMaybe[A, S](maybe: => Maybe[A] < S): A < (S & Abort[Maybe.Empty]) =
+        maybe.map(m => Abort.get(m.toRight[Maybe.Empty](Maybe.Empty)))
```

Comment:
You can directly invoke `Abort.get`:
https://github.com/getkyo/kyo/blob/main/kyo-prelude/shared/src/main/scala/kyo/Abort.scala#L52-L54

Reply:
Missed that -- thanks!